platform/default/default_file_source.cpp

```diff
@@ -132,7 +123,7 @@ class DefaultFileSource::Impl {
 
     OfflineDatabase offlineDatabase;
     OnlineFileSource onlineFileSource;
-    std::unordered_map<AsyncRequest*, std::unique_ptr<Task>> tasks;
+    std::unordered_map<AsyncRequest*, std::unique_ptr<AsyncRequest>> tasks;
```

Comment:
Per chat: using `unordered_map` is usually preferable when you don't need ordering because search, insertion, and removal are amortized constant time complexity, compared to `map`'s logarithmic complexity.